### PR TITLE
Allow 0-dimensional inputs for var_mean

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Reducing 0-dimensional inputs is allowed since https://github.com/csarofeen/pytorch/pull/1771.
In this PR I added a fallback path for the `var_mean` op, so that it doesn't hit this check https://github.com/NVIDIA/Fuser/blob/4483f51a396a8c594caae8078e3cb82c7d3caa44/csrc/ops/arith.cpp#L1724
The change in the `numFeatures` function is required to avoid segfaults.

### CODE REVIEW

File: csrc/ops/normalization.cpp [hunk 1]
```diff
@@ -115,6 +118,11 @@ VarMeanResult variance_mean(
         sub(num_features, IrBuilder::create<Int>(x->container(), correction));
   }
 
+  // Welford op can't handle 0-dim tensors, so we need to handle them separately
+  if (x->nDims() == 0) {
+    return {variance(x, dims, correction, keepdim), mean(x, dims, keepdim)};
```
Comment (jjsjann123) [inline csrc/ops/normalization.cpp:123]: Feels a little bit awkward here, since variance for a scalar tensor is just `nan` and mean is just returning itself. Having said that, we might want to keep it like this in the definition, that feels like an optimization pass's job to simplify that.

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: csrc/ops/normalization.cpp (hunk 1 above)
Comment (IvanYashchuk) [reply csrc/ops/normalization.cpp:123]: I agree, but functions in `FusidionDefinition.ops` should still mostly work with edge cases like that. Otherwise, a higher-level project using these functions would need to have more and more wrappers that fix problematic cases.

Comment (IvanYashchuk) [reply csrc/ops/normalization.cpp:123]: It's not just `nan`, it's also `0.0` 😄 
```py
In [1]: import torch

In [2]: a = torch.randn(())

In [3]: torch.var(a, dim=0, correction=0)
Out[3]: tensor(0.)

In [4]: torch.var(a, dim=0, correction=1)
Out[4]: tensor(nan)
```

**Comment by csarofeen** [comment]
@IvanYashchuk any reason you didn't merge this in?

**Comment by jjsjann123** [comment]
!test

File: tests/cpp/test_gpu2.cpp [hunk 2]
```diff
@@ -2674,6 +2674,36 @@ TEST_F(NVFuserTest, FusionVarMean_CUDA) {
   }
 }
 
+TEST_F(NVFuserTest, FusionVarMeanZeroDim_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  int correction = 0;
+  bool keepdim = false;
+
```
Comment (xwang233) [inline tests/cpp/test_gpu2.cpp:2684]:
```suggestion
  at::ScalarType dtype = at::kFloat;  // Declare dtype before use
```
**[Suggested by AI]** One line was added to declare and initialize the variable `dtype`:

+  at::ScalarType dtype = at::kFloat;  // Declare dtype before use

This ensures that `dtype` is defined (as a float scalar type) before it is referenced elsewhere in the code, eliminating the previous build error caused by an undeclared or uninitialized variable.

**Comment by jjsjann123** [comment]
!test
